frontend/src/container/Favorites.jsx

```diff
@@ -0,0 +1,292 @@
+  removeFavorite,
+  setHeadTitle
+} from '../action-creator.sync.js'
+
+import {
+  getFavoriteContentList,
+  removeContentFromFavoriteList
+} from '../action-creator.async.js'
+import {
+  FETCH_CONFIG
+} from '../util/helper.js'
+
+import ContentListItem from '../component/ContentListItem.jsx'
+import ContentType from '../component/ContentType.jsx'
+
+require('../css/Favorites.styl')
+
+const FavoritesHeader = translate()(props => {
+  return (
+    <div className='favoritesHeader content__header'>
+      <div className='favoritesHeader__type'>
+        {props.t('Type')}
+      </div>
+      <div className='favoritesHeader__title'>
+        {props.t('Title and path')}
+      </div>
+      {/* Header for windows smaller than max-sm */}
+      <div className='favoritesHeader__title-max-sm'>
+        {props.t('Title')}
+      </div>
+      <div className='favoritesHeader__modification'>
+        {props.t('Last Modification')}
+      </div>
+      <div className='favoritesHeader__information'>
+        {props.t('Information_plural')}
+      </div>
+      <div className='favoritesHeader__favoriteButton'>
+        {props.t('Favorite')}
+      </div>
+    </div>
+  )
+})
+
+const UnavailableContent = translate()(props => {
+  return (
+    <ListItemWrapper
+      label={props.label}
+      read
+      contentType={props.contentTypeInfo}
+      isLast={props.isLast}
+      customClass='unavailableContent contentListItem'
+    >
+      <ContentType
+        contentTypeInfo={props.contentTypeInfo}
+        customClass='contentListItem__type'
+      />
+      <div className='contentListItem__name_path unavailableContent__name_warning'>
+        {props.label}
+        <span className='unavailableContent__warning'>
+          <Icon
+            icon='fa-fw fas fa-exclamation-triangle'
+            title={props.t('Warning')}
+          />
+          &nbsp;
+          {props.t('content is not available')}
+        </span>
+      </div>
+      {props.children}
+    </ListItemWrapper>
+  )
+})
+
+export class Favorites extends React.Component {
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      contentCommentsCountList: [],
+      contentBreadcrumbsList: []
+    }
+
+    props.registerCustomEventHandlerList([
+      { name: CUSTOM_EVENT.ALL_APP_CHANGE_LANGUAGE, handler: this.handleAllAppChangeLanguage }
+    ])
+  }
+
+  handleAllAppChangeLanguage = () => {
+    this.setHeadTitle()
+    this.buildBreadcrumbs()
+  }
+
+  setHeadTitle = () => {
+    const { props } = this
+    const headTitle = buildHeadTitle([props.t('Favorites')])
+    props.dispatch(setHeadTitle(headTitle))
+  }
+
+  buildBreadcrumbs = () => {
+    const { props } = this
+
+    props.dispatch(setBreadcrumbs([{
+      link: PAGE.FAVORITES,
+      type: BREADCRUMBS_TYPE.CORE,
+      label: props.t('Favorites'),
+      isALink: true
+    }]))
+  }
+
+  componentDidMount () {
+    this.setHeadTitle()
+    this.buildBreadcrumbs()
+    this.loadFavoriteList()
+  }
```

Comment:
This makes me think that we should probably have a component that automatically does this stuff (not for this PR)

Reply:
Yes, I also had the idea, the boilerplate code is a bit big to my taste.

Reply:
https://github.com/tracim/tracim/issues/4397